plugins/TimingTriggerCandidateMaker.cpp

```diff
@@ -37,48 +38,87 @@ TimingTriggerCandidateMaker::TimingTriggerCandidateMaker(const std::string& name
   register_command("scrap", &TimingTriggerCandidateMaker::do_scrap);
 }
 
-triggeralgs::TriggerCandidate
+template<typename T> 
+std::vector<int> 
+TimingTriggerCandidateMaker::GetTriggeredBits(T signal_map)
+{
+  std::vector<int> triggered_bits;
+  int nbits = std::numeric_limits<T>::digits;
+  for (int i = 0; i < nbits; ++i) {
+    if ((signal_map >> i) & 1 ) {
```

Comment:
I had to do a similar (same?) thing for the CTB plugin, using bitset instead of vector and looping over (see: https://github.com/DUNE-DAQ/trigger/blob/mrigan/ctb_integration/plugins/CTBTriggerCandidateMaker.cpp#L49C3-L49C40). Just another implementation possibility if you'd like.

Reply:
Initially I had something pretty much identical, but decided to abstract out to not have the hard coded 32 and keep the rest semi-clean

Reply:
It seems slower to me from a glance, but I have not tested/compared the implementations. I'm happy with both though.